.github/workflows/master_check_approval.yml

```diff
@@ -0,0 +1,78 @@
+---
+name: Check Required Approvals
+on:
+  workflow_call:
+    inputs:
+      required-teams:
+        description: 'Teams required for approval in the PR'
+        required: true
+        type: string
+ 
+jobs:
+
+  check-approval:
+    runs-on: non-build
+    steps:
+        - name: Checkout code
+          uses: actions/checkout@v3
+
+        - name: Dump GitHub context
+          env:
+            GITHUB_CONTEXT: ${{ toJson(github) }}
+          run: |
+            echo "$GITHUB_CONTEXT"
+
+        - name: Check commits
+          id: get-pr-commits
+          run: |
+            PR_COMMITS=$(curl -L \
+              -H "Accept: application/vnd.github+json" \
+              -H "Authorization: Bearer ${{ secrets.CIPIPELINE_NEXUS_PASSWORD }}" \
+              -H "X-GitHub-Api-Version: 2022-11-28" \
+              https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.pull_request.number }}/commits| jq -r '. | length')
+            echo "Number of commits in PR: $PR_COMMITS"
+            echo "::set-output name=pr_commits::$PR_COMMITS"
+
+        - name: Check Approval
+          env:
+            TOKEN: ${{ secrets.CIPIPELINE_NEXUS_PASSWORD }}
+          run: |
+            title="${{github.event.pull_request.head.ref}}"
+            user="${{github.event.pull_request.user.login}}"
+            if [[ "$title" == *"double_commit"* && "$user" == "cipipelinetessell" ]]; then
+              PR_COMMITS=${{ steps.get-pr-commits.outputs.pr_commits }}
+              if [ "$PR_COMMITS" -eq 1 ]; then
+                echo "One Commit present in DC."
+                exit 0
+              else
+                echo "More than one commit present in DC, need approval."
+              fi
+            fi
+            ORG_NAME="TessellDevelopment"
+            REPO="${{github.repository}}"
+            PR_NUMBER="${{ github.event.pull_request.number }}"
+            TEAMS="${{inputs.required-teams}}"
+            IFS=' ' read -ra TEAMS_ARRAY <<< "$TEAMS"
+            for TEAM in "${TEAMS_ARRAY[@]}"; do
+              FLAG=0
+              MEMBERS=$(curl -s -H "Authorization: Bearer $TOKEN" "https://api.github.com/orgs/$ORG_NAME/teams/$TEAM/members" | jq -r '.[].login')
+              REVIEWERS=$(curl -L \
+                -H "Accept: application/vnd.github+json" \
+                -H "Authorization: Bearer $TOKEN" \
+                -H "X-GitHub-Api-Version: 2022-11-28" \
+                https://api.github.com/repos/$REPO/pulls/$PR_NUMBER/reviews | jq -r '.[] | select(.state == "APPROVED") | .user.login')
+              IFS=, read -r -a REVIEWER_ARRAY <<< "$(echo -e "$REVIEWERS" | tr '\n' ',')"
+              IFS=, read -r -a MEMBER_ARRAY <<< "$(echo -e "$MEMBERS" | tr '\n' ',')"
+              for REVIEWER in "${REVIEWER_ARRAY[@]}"; do
```

Comment:
I see we are not considering the rejection. If one member rejects, then PR must not be allowed to merge.